.github/action/setup-node-env/action.yml

```diff
@@ -0,0 +1,53 @@
+# this is a reusable workflow that will be called by the
+# push_staging and push_production workflows
+# more info: https://docs.github.com/en/actions/learn-github-actions/reusing-workflows
+
+name: gutenberg/setup-node-env
+description: Setup the Node.js environment including `pnpm` and dependencies
+
+inputs:
+    install:
+        default: true
+        description: Whether to install dependencies.
+    install-working-directory:
+        required: false
+        description: Optional working directory to run installation in.
+    frozen-lockfile:
+        default: false
+        description: Whether to install dependencies using `--frozen-lockfile`.
+    node-version:
+        default: 14
```

Comment:
I think this is nice that we don't have to repeat passing `node-version` in every workflow if it's just running the default version. We can just update this file when upgrading node to a major version.

Reply:
I guess this could also be changed to `node-version-file: '.nvmrc'` so that https://github.com/WordPress/gutenberg/blob/trunk/.nvmrc is the only place to change the version.

https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file

Reply:
Nice find! That sounds like a good option 🎉 We'd need to keep around `node-version` still for the workflows that use a node version matrix but referring to the `.nvmrc` file for the default is 💯

Reply:
`.nvmrc ` plays an important role for branches where we apply fixes to older WordPress major releases like `wp/5.1`.

Reply:
Okay in https://github.com/WordPress/gutenberg/pull/38624/commits/bd1d499c05c0a01cae5b3c157ca549130f551102 I tried to set it up to just use `.nvmrc` by default unless a specific `node-version` is passed in (to preserve the ability for workflows to use the node version matrix, if that's still necessary). Hopefully it works, I'm not too strong on GitHub action expressions but should only need a little tweaking if it doesn't work right off the bat.